malloc/malloc.c

```diff
@@ -13,7 +13,14 @@
 */
 char *string_dup(char *src)
 {
-
+   int size = string_length(src);
+   char *output = malloc(size + 1);
+   for (int i = 0; i < size; i++){
+       output[i] = src[i];
+   }
+   output[size] = '\0';
+
+   return output;
```

Comment:
Just for fun, remember you made a `string_copy` function? You could import that function here and actually use it, and it'd work swimmingly here!